app/controllers/admin/routes.py

```diff
@@ -154,6 +154,10 @@ def eventDisplay(eventId):
         eventData['timeStart'] = event.timeStart.strftime("%-I:%M %p")
         eventData['timeEnd'] = event.timeEnd.strftime("%-I:%M %p")
         eventData["startDate"] = event.startDate.strftime("%m/%d/%Y")
+        # List below is to identify the last event in the series
+        eventSeriesList = list(Event.select().where(Event.recurringId == event.recurringId))
+        if event.recurringId and not eventSeriesList[-1] == event:
+            eventData["nextRecurringEvent"] = Event.select().where(Event.id > event.id, Event.recurringId == event.recurringId).get()
```

Comment:
You already have the list of recurring events, you don't need to select from the database again to find the next one. Try `list.index()` here, and for your conditional